### PR TITLE
runk: Return error when tty is used without console socket

### PR DESCRIPTION
**Pull request by ManaSugi**
runk always launches containers with detached mode,
so users have to use a console socket with `run` and
`create` operations when a terminal is used.
If users set `terminal` to `true` in `config.json` and
try to launch a container without specifying a console
socket, runk returns an error with a message early.

Fixes: #4324

Signed-off-by: Manabu Sugimoto <Manabu.Sugimoto@sony.com>

### CODE REVIEW

File: src/tools/runk/libcontainer/src/builder.rs [hunk 1]
```diff
@@ -48,6 +48,16 @@ impl Container {
             }
         }
 
+        if let Some(process) = spec.process.as_ref() {
+            // runk always launches containers with detached mode, so users have to
+            // use a console socket with run or create operation when a terminal is used.
+            if process.terminal && self.console_socket.is_none() {
+                return Err(anyhow!(
+                    "cannot allocate a pseudo-TTY without setting a console socket"
```
Comment (jodh-intel) [inline src/tools/runk/libcontainer/src/builder.rs:56]: Could you add a unit test for this scenario?

Comment (ManaSugi) [reply src/tools/runk/libcontainer/src/builder.rs:56]: @jodh-intel Thank you for the review. I added the unit test.

**Review by liubin — APPROVED** [review]

**Pull request merged**